### PR TITLE
Greengrass Implement create_core_definition

### PR DESCRIPTION
**Pull request by cm-iwata**
Implement Greengrass create_core_definition

### CODE REVIEW

File: moto/greengrass/models.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        core_definition = FakeCoreDefinition(self.region_name, name)
+        self.core_definitions[core_definition.id] = core_definition
+        self.create_core_definition_version(
+            core_definition.id, initial_version["Cores"]
+        )
+        return core_definition
+
+    def create_core_definition_version(self, core_definition_id, cores):
+
+        definition = {"Cores": cores}
```
Comment (cm-iwata) [inline moto/greengrass/models.py:84]: If we want to check strictly, we need to check IoT thing and certificate exist in the IoT backend.
However, other modules like awslambda are skipping similar checks, so I'm skipping checks here as well.

File: moto/greengrass/responses.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    def create_core_definition(self):
+        name = self._get_param("Name")
+        initial_version = self._get_param("InitialVersion")
+        res = self.greengrass_backend.create_core_definition(
+            name=name, initial_version=initial_version
+        )
+        return 201, {"status": 201}, json.dumps(res.to_dict())
+
+    def create_core_definition_version(self):
```
[the remark this answers is not in the document]
Comment (cm-iwata) [reply moto/greengrass/responses.py:22]: @bblommers 
Sorry I forgot.
I added test for create_core_definition_version

**Comment by bblommers** [comment]
Thanks for adding the test @cm-iwata. Hope you don't mind - I made a few changes to ensure that the tests pass in ServerMode. All the tests should pass now (:crossed_fingers: ..)

**Review by bblommers — APPROVED** [review]
Looks like that did the trick. Thanks again for the PR @cm-iwata!

**Pull request merged**